.github/workflows/master.yml

```diff
@@ -462,6 +463,42 @@ jobs:
         - 14.x
         pythonversion:
         - "3.7"
+  generate_coverage_data:
+    name: generate_coverage_data
+    needs: prerequisites
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+    - name: Generate Coverage Data
+      run: |
+        echo "Coverage output directory: $COVERAGE_OUTPUT_DIR"
+        make tfgen
+  upload_coverage_data:
+    name: upload_coverage_data
+    needs: generate_coverage_data
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+    - name: Obtain current date & S3 address
+      run: |
+        currDateTime=$(date +"%Y-%m-%d_%H-%M-%S")
+        newSummaryName="summary_${currDateTime}.json"
```

Comment:
Remember it won't be just this provider uploading summary docs - there will be others so we should ensure that we prefix the file with the name of the provider